metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/AlertRuleEvaluator.java

```diff
@@ -0,0 +1,86 @@
+package org.hypertrace.alert.engine.metric.anomaly.detector.evaluator;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Optional;
+import org.hypertrace.alert.engine.eventcondition.config.service.v1.MetricAnomalyEventCondition;
+import org.hypertrace.alert.engine.eventcondition.config.service.v1.ViolationCondition;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.AlertTask;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.NotificationEvent;
+import org.hypertrace.core.attribute.service.client.AttributeServiceClient;
+import org.hypertrace.core.query.service.client.QueryServiceClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AlertRuleEvaluator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AlertRuleEvaluator.class);
+  static final String METRIC_ANOMALY_ACTION_EVENT_TYPE = "MetricAnomalyViolation";
```

Comment:
Q: Used somewhere else, can be `private`?

Reply:
yeah used somewhere else